library/ssl_tls.c

```diff
@@ -3665,7 +3735,13 @@ void mbedtls_ssl_handshake_free( mbedtls_ssl_context *ssl )
     mbedtls_ecdh_free( &handshake->ecdh_ctx );
 #endif
 #if defined(MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED)
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    psa_pake_abort( &handshake->psa_pake_ctx );
```

Comment:
Don't we want to check `ssl->handshake->psa_pake_ctx_is_ok`? Is everything going to be handled fine if `mbedtls_ssl_set_hs_ecjpake_password` fails and then `mbedtls_ssl_handshake_free` will be called?
Is there a test for it?

Reply:
According to [here](https://armmbed.github.io/mbed-crypto/1.1_PAKE_Extension.0-bet.0/html/pake.html#c.psa_pake_abort) it seems that;

> This function can be called any time after the operation object has been initialized as described in [psa_pake_operation_t](https://armmbed.github.io/mbed-crypto/1.1_PAKE_Extension.0-bet.0/html/pake.html#c.psa_pake_operation_t)

I don't know if there's a test for this, but I can try to take a look

Reply:
I don't think we want to check `ssl->handshake_psa_pake_ctx_is_ok` - as Valerio says, `abort()` is supposed to be safe to call, so it's not a problem to call it even when it isn't strictly needed. If we don't have one already, we might want to add a test in `test_suite_psa_crypto_pake()` ensuring that calling `abort()` twice in a row works. But I think adding tests at the TLS level is pretty low priority.